src/include/daos_types.h

```diff
@@ -257,6 +258,51 @@ daos_is_valid_uuid_string(const char *uuid)
 
 	return true;
 }
+
+static inline bool
+daos_is_valid_uuid_lower_string(const char *uuid)
```

Comment:
It doesn't look like this version includes all the changes that landed to master, which combined some of this logic. Might be easier to review to cherry-pick of the final version of the master change (either by rebasing or opening a new PR).